homeassistant/components/generic/config_flow.py

```diff
@@ -399,3 +434,39 @@ async def async_step_init(
             ),
             errors=errors,
         )
+
+
+class CameraImagePreView(CameraImageView):
+    """Camera view to temporarily serve an image."""
+
+    name = "api:generic:preview_flow_image"
+    previews: dict[str, GenericCamera] = {}
+    initialized = False
+
+    def __init__(
```

Comment:
(1) It is not clear to me that inheriting from CameraImageView is giving much value on top of just implementing a stanalone view here. 
(2) The more common model for a view is to have a single object of the view registered, then have the request parameters passed in to perform some sort of lookup on the enttiy or hass.data object, etc.  You could, for example, put the flow id as a url parameter here and grab the object off the flow

Reply:
I realize (2) is a similar comment to above about taking the flow id as a path parameter.

Reply:
Oh also see (2) is also the same as the comment above about registering the view once.

Reply:
I think it is already working as you describe.

First time the user does this, the view is created. Second time onwards it is just adding a preview_id to a dict so that it knows what to display when a request comes in.

I originally had it registering one view per flow, but that was discouraged in the first review comment.

Not sure what I need to change.

Reply:
(1) Is don't inherit CameraImageView (i see baloob left similar comment in parallel)
(2) (a) The camera view should be register in, say, `async_setup` rather than during the middle of the config flow. that setup in the config flow can run multiple times in the lifetime of home assistant, creating multiple instances of `CameraImagePreView`

Reply:
Ok, I will have another look tomorrow evening.

I think I get what you mean.

Thanks for the feedback!

Reply:
@allenporter Ok, I get what you mean.  Although I'm not sure where to put that code.  `async_setup()` doesn't seem to run until you've got a valid config.  Is there another hook location I can use to get the view to register?

Reply:
Ah ok, i was worried about that. I am not aware of another hook to use, so maybe then just have a separate method that does the registration and stores a bit in hass.data[DOMAIN] to ensure it only does it once?

Reply:
@allenporter Updated - is this what you had in mind?

Reply:
Yeah, exactly. Maybe someone else knows how to do this better, but that's the best i can think of? Thanks!